### PR TITLE
Feature 264 - Metrics: Add advertising metrics to ExperienceEvent

### PR DESCRIPTION
**Pull request by gabebarcelos-adobe**
PR to add AdCloud's advertising metrics to ExperienceEvent. This is related to open issue #264.
@kstreeter @cdegroot-adobe

### CODE REVIEW

File: schemas/data/conversions.schema.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/conversions",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "conversions",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/metric"
+  ],
+  "description": "The advertising metric conversions indicates a predefined conversion event is triggered by a surfer online",
```
Comment (cdegroot-adobe) [inline schemas/data/conversions.schema.json]: Is a conversion not analysis specific. The observed events may show that the ad was consumed to 75%. Maybe today that is considered a conversion, but tomorrow it is considered a conversion only when it hits 90%.

File: schemas/data/first-quartiles.schema.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/first-quartiles",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "first-quartiles",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": ["https://ns.adobe.com/xdm/data/metric"],
+  "description": "The advertising metric first-quartiles describes digital video ads which played continuously for at least 25% of the total duration at normal speed",
```
Comment (cdegroot-adobe) [inline schemas/data/first-quartiles.schema.json]: "played continuously"  seems quite specific. Does it mean no pauses, either human or buffering for example, or does it mean the first 25% was watched "somehow"?

File: schemas/data/intrumented-impressions.schema.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/viewability/intrumented-impressions",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "intrumented-impressions",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/metric"
+  ],
+  "description": "The advertising/viewability metric intrumented-impressions describes the whether the video impression had viewabilty library instrumented.",
```
Comment (cdegroot-adobe) [inline schemas/data/intrumented-impressions.schema.json]: There is a typo, remove "the" after describes. 
I am not keen on the instrumented vs non instrumented nature of this. Can we represent this as an impression, and have how it was viewed as a property elsewhere?

File: schemas/data/measured-ad-not-visible.schema.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/viewability/measured-ad-not-visible",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "measured-ad-not-visible",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/metric"
+  ],
+  "description": "The advertising/viewability metric measured-ad-not-visible describes the whether < 51% of pixels on an ad where visible during playtime.",
```
Comment (cdegroot-adobe) [inline schemas/data/measured-ad-not-visible.schema.json]: Can this be a property of the quantiles metric?

Comment (cdegroot-adobe) [reply schemas/data/measured-ad-not-visible.schema.json]: Typo, move "the" before "whether" to after "whether"

File: schemas/data/measured-muted.schema.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/viewability/measured-muted",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "measured-muted",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/metric"
+  ],
+  "description": "The advertising/viewability metric measured-muted describes the whether at start time the ad was muted.",
```
Comment (cdegroot-adobe) [inline schemas/data/measured-muted.schema.json]: Typo, remove "the" before "whether"

Comment (cdegroot-adobe) [reply schemas/data/measured-muted.schema.json]: Can this not be a property of the quantiles metric?

File: schemas/data/measured-window-inactive.schema.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "$id": "https://ns.adobe.com/xdm/data/metrics/advertising/viewability/measured-window-inactive",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "measured-window-inactive",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/data/metric"
+  ],
+  "description": "The advertising/viewability metric measured-window-inactive describes the whether at impression time the ad was displayed in a window not active by the user.",
```
Comment (cdegroot-adobe) [inline schemas/data/measured-window-inactive.schema.json]: Typo, remove "the" before "whether"

Can this not be a property of the quantiles metric?

File: schemas/data/metrics.schema.json [hunk 7]
```diff
@@ -42,25 +42,88 @@
           "$ref": "https://ns.adobe.com/xdm/data/measure",
           "description": "Graceful termination(s) of an application."
         },
+        "https://ns.adobe.com/xdm/data/metrics/advertising/completes": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description": "A digital video ad has played through it's entirety."
+        },
+        "https://ns.adobe.com/xdm/data/metrics/advertising/conversion/conversions": {
```
Comment (cdegroot-adobe) [inline schemas/data/metrics.schema.json]: `...\conversion\conversions` can it be `...\conversions`

**Comment by lrosenthol** [comment]
One concern here is that “conversion metrics” in DC refers to the number of files that have been converted (from one format to another), how quickly, etc.  
However, it looks like this is all inside an extension schema, so that should be OK

File: schemas/data/conversions.schema.json (hunk 1 above)
Comment (gabebarcelos-adobe) [reply schemas/data/conversions.schema.json]: Yes, you are right in a way, however this metric is describing/counting the trigger event itself that carries the metadata leading to a Conversion. There are post-event processes which look over all these Experiences Events and either attribute or not this event to a real world Conversion stitching together different events in a path to conversion.

File: schemas/data/first-quartiles.schema.json (hunk 2 above)
Comment (gabebarcelos-adobe) [reply schemas/data/first-quartiles.schema.json]: Cleaned up description.

File: schemas/data/conversions.schema.json (hunk 1 above)
Comment (cdegroot-adobe) [reply schemas/data/conversions.schema.json]: Ok, got it.

**Comment by cdegroot-adobe** [comment]
@kstreeter  @trieloff can you please review this as soon as possible. Thanks C.

**Comment by trieloff** [comment]
@gabebarcelos-adobe please add a line to `CONTRIBUTING.md` explaining that `https://ns.adobe.com/xdm/data/metrics/advertising/impressions` has been renamed. Otherwise it looks good to me.

**Comment by gabebarcelos-adobe** [comment]
@trieloff - https://ns.adobe.com/xdm/data/metrics/advertising/impressions should not have been renamed, we just added additional metrics.

**Comment by jwen-adobe** [comment]
@gabebarcelos-adobe I removed "type": "object" from field "xdm:implementationDetails" to avoid validation warning.

**Pull request merged**